src/main/scala/beam/agentsim/infrastructure/parking/ParkingNetwork.scala

```diff
@@ -0,0 +1,86 @@
+package beam.agentsim.infrastructure.parking
+
+import beam.agentsim.Resource.ReleaseParkingStall
+import beam.agentsim.agents.vehicles.VehicleManager
+import beam.agentsim.infrastructure._
+import beam.utils.metrics.SimpleCounter
+import com.typesafe.scalalogging.LazyLogging
+import org.matsim.api.core.v01.Id
+
+abstract class ParkingNetwork[GEO: GeoLevel](
+  vehicleManagerId: Id[VehicleManager],
+  parkingZones: Map[Id[ParkingZoneId], ParkingZone[GEO]]
+) extends LazyLogging {
+
+  // Generic
+  protected val searchFunctions: Option[InfrastructureFunctions[_]]
+
+  // Core
+  protected var totalStallsInUse: Long = 0L
+  protected var totalStallsAvailable: Long = parkingZones.map(_._2.stallsAvailable).sum
+
+  def getVehicleManagerId: Id[VehicleManager] = vehicleManagerId
+
+  /**
+    *
+    * @param inquiry ParkingInquiry
+    * @param parallelizationCounterOption Option[SimpleCounter]
+    * @return
+    */
+  def processParkingInquiry(
+    inquiry: ParkingInquiry,
+    parallelizationCounterOption: Option[SimpleCounter] = None
+  ): Option[ParkingInquiryResponse] = {
+    logger.debug("Received parking inquiry: {}", inquiry)
+    searchFunctions.map(_.searchForParkingStall(inquiry)).getOrElse(None) match {
+      case Some(_ @ParkingZoneSearch.ParkingZoneSearchResult(parkingStall, parkingZone, _, _, _)) =>
```

Comment:
This can be a map - as a none results in a none there anyway. And it does not appear to be doing anything special in the first match beyond pulling out data.

Reply:
Correct!